modules/monitoring-adding-a-secret-to-an-alertmanager-configuration.adoc

```diff
@@ -0,0 +1,95 @@
+
+:_content-type: PROCEDURE
+[id="monitoring-adding-a-secret-to-an-alertmanager-configuration_{context}"]
+= Adding a secret to an Alertmanager configuration
+
+You can add secrets to the Alertmanager configuration for core platform monitoring components by editing the `cluster-monitoring-config` config map for the `openshift-monitoring` project
+You can add secrets to the Alertmanager configuration for user-defined projects by editing the `user-workload-monitoring-config` config map in the `openshift-user-workload-monitoring` project.
+
+.Prerequisites
+
+* You have installed the OpenShift CLI (`oc`).
+* You have created the secret for the endpoint receiver.
+* *If you are configuring core {product-title} monitoring components in the `openshift-monitoring` project*:
+** You have access to the cluster as a user with the `cluster-admin` role.
+** You have created the `cluster-monitoring-config` config map.
+* *If you are configuring components that monitor user-defined projects*:
+** A cluster administrator has enabled monitoring for user-defined projects
+** You have access to the cluster as a user with the `cluster-admin` role, or as a user with the `user-workload-monitoring-config-edit` role in the `openshift-user-workload-monitoring` project.
+** You have created the `user-workload-monitoring-config` config map.
+
+.Procedure
+
+. To add a secret configuration to Alertmanager for core platform monitoring, edit the `cluster-monitoring-config` config map in the `openshift-monitoring` project:
++
+[source,terminal]
+----
+$ oc -n openshift-monitoring edit configmap cluster-monitoring-config
+----
+
+. Add a `secrets:` section under `data/config.yaml/alertmanagerMain`.
+
+. Add the configuration details for the secret in this section:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    alertmanagerMain:
+      secrets:
+      - <secret_name> <1>
+      - <secret_receiver_credentials> <2>
+----
+<1> The name of the `Secret` object that contains the authentication credentials.
+<2> The key or keys in the specified `Secret` object that contain the authentication credentials.
++
+The following sample config map settings configure Alertmanager to use a secret named `test-secret` that authenticates using a token in the `test-receiver-token` key in the `Secret` object:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    alertmanagerMain:
+      secrets:
+      - test-secret
+      - test-receiver-token
+----
+
+. Optional: To configure Alertmanager for user-defined projects to reference the same secret, use the same configuration, but instead add the secret name and key information under `data/config.yaml/alertmanager` in the `user-workload-monitoring-config` config map in the `openshift-user-workload-monitoring` project:
```

Comment:
just a reminder here, to configure Alertmanager for user-defined projects to reference the same secrets, the secrets should be both created under openshift-monitoring and openshift-user-workload-monitoring projects, since the secret resources is restricted to project

Reply:
Updated to reflect this behavior